src/webui/api/appcontroller.cpp

```diff
@@ -1172,6 +1181,63 @@ void AppController::getDirectoryContentAction()
     setResult(ret);
 }
 
+void AppController::cookiesAction()
+{
+    const QList<QNetworkCookie> &cookies = Net::DownloadManager::instance()->allCookies();
+    QJsonArray ret;
+    for (const QNetworkCookie &cookie : cookies)
+    {
+        ret << QJsonObject{
+            {KEY_COOKIE_NAME, QString::fromLatin1(cookie.name())},
+            {KEY_COOKIE_DOMAIN, cookie.domain()},
+            {KEY_COOKIE_PATH, cookie.path()},
+            {KEY_COOKIE_VALUE, QString::fromLatin1(cookie.value())},
+            {KEY_COOKIE_EXPIRATION_DATE, Utils::DateTime::toSecsSinceEpoch(cookie.expirationDate())},
+        };
+    }
+
+    setResult(ret);
+}
+
+void AppController::setCookiesAction()
+{
+    requireParams({u"cookies"_s});
+    const QString cookiesParam {params()[u"cookies"_s].trimmed()};
+
+    QJsonParseError jsonError;
+    const auto cookiesJsonDocument = QJsonDocument::fromJson(cookiesParam.toUtf8(), &jsonError);
+    if (jsonError.error != QJsonParseError::NoError)
+        throw APIError(APIErrorType::BadParams, tr("unable to parse"));
+    if (!cookiesJsonDocument.isArray())
+        throw APIError(APIErrorType::BadParams, tr("cookies must be array"));
+
+    const QJsonArray &cookiesJsonArr = cookiesJsonDocument.array();
```

Comment:
```suggestion
    const QJsonArray cookiesJsonArr = cookiesJsonDocument.array();
```

Reply:
Same question here - why don't we want a reference?